### PR TITLE
Core: Make Iceberg version in EnvironmentContext by default

### PR DESCRIPTION
**Pull request by wangyinsheng**
Make Iceberg version in EnvironmentContext by default

### CODE REVIEW

File: core/src/test/java/org/apache/iceberg/TestEnvironmentContext.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+public class TestEnvironmentContext {
+
+  @Test
+  public void testDefaultValue() {
+    assertEquals(
```
[the remark this answers is not in the document]
Comment (wangyinsheng) [reply core/src/test/java/org/apache/iceberg/TestEnvironmentContext.java]: done

File: core/src/main/java/org/apache/iceberg/EnvironmentContext.java [hunk 2]
```diff
@@ -32,6 +32,10 @@ private EnvironmentContext() {
 
   private static final Map<String, String> PROPERTIES = Maps.newConcurrentMap();
 
+  static {
+    PROPERTIES.put("iceberg-version", IcebergBuild.fullVersion());
```
[the remark this answers is not in the document]
Comment (wangyinsheng) [reply core/src/main/java/org/apache/iceberg/EnvironmentContext.java:34]: done

**Review by nastra — APPROVED** [review]
I don't recall why I put this into the constructor in the first place, so this LGTM

**Comment by nastra** [comment]
@Fokko could you review this one please and also kick off CI?

**Comment by wangyinsheng** [comment]
@Fokko PTAL

**Comment by wangyinsheng** [comment]
@rdblue PTAL

**Comment by wangyinsheng** [comment]
@nastra @Fokko @rdblue  Hi, please take a look this pr,  This is an improvement on https://github.com/apache/iceberg/pull/6185  , and it's my first pr,  It's been going on for a week. Thanks a lot

**Review by Fokko — APPROVED** [review]
Thanks @wangyinsheng for pinging me. I was a bit distracted with other work. LGTM!

Thanks @nastra for the review!

**Pull request merged**